include/picongpu/fields/currentDeposition/EmZ/EmZ.hpp

```diff
@@ -123,8 +118,47 @@ namespace picongpu
                         line.m_pos1[d] = calc_InCellPos(posEnd[d], I[1][d]);
                         line.m_pos0[d] = calc_InCellPos(relayPoint[d], I[1][d]);
                     }
-                    deposit(acc, dataBoxJ.shift(I[1]).toCursor(), line, chargeDensity, velocity.z() * float_X(0.5));
+                    deposit(acc, dataBoxJ.shift(I[1]).toCursor(), line, chargeDensity);
                 }
+
+                /* 2d case requires special handling of Jz as explained in #3889.
+                 * Pass dataBoxJ as auto&& to defer evaluation for 3d case.
+                 */
+                pmacc::meta::invokeIf<simDim == 2>(
+                    [&](auto&& dataBoxJ)
```

Comment:
```diff
diff --git a/include/picongpu/fields/currentDeposition/EmZ/EmZ.hpp b/include/picongpu/fields/currentDeposition/EmZ/EmZ.hpp
index 8e2fff17f..125490eaa 100644
--- a/include/picongpu/fields/currentDeposition/EmZ/EmZ.hpp
+++ b/include/picongpu/fields/currentDeposition/EmZ/EmZ.hpp
@@ -125,7 +125,7 @@ namespace picongpu
                  * Pass dataBoxJ as auto&& to defer evaluation for 3d case.
                  */
                 pmacc::meta::invokeIf<simDim == 2>(
-                    [&](auto&& dataBoxJ)
+                    [&, this](auto&& dataBoxJ)
                     {
                         /* For Jz we consider the whole movement on a step.
                          * This movement it is not necessarily on support.
@@ -140,8 +140,8 @@ namespace picongpu
                          */
                         for(uint32_t d = 0; d < simDim; ++d)
                         {
-                            line.m_pos0[d] = calc_InCellPos(posStart[d], I[1][d]);
-                            line.m_pos1[d] = calc_InCellPos(posEnd[d], I[1][d]);
+                            line.m_pos0[d] = this->calc_InCellPos(posStart[d], I[1][d]);
+                            line.m_pos1[d] = this->calc_InCellPos(posEnd[d], I[1][d]);
                         }
                         /* Have to use DIM2, otherwise 3d case wouldn't compile due to
                          * no computeCurrentZ() method.
```

Is required to pass the CI.

The current error is:

```
/picongpu/include/pmacc/../picongpu/simulation/stage/CurrentDeposition.hpp:79:52:   required from here
/picongpu/include/pmacc/../picongpu/fields/currentDeposition/EmZ/EmZ.hpp:143:60: error: 'calc_InCellPos' was not declared in this scope
                             line.m_pos0[d] = calc_InCellPos(posStart[d], I[1][d]);
```